### PR TITLE
completed ticket 5

### PR DESCRIPTION
**Pull request by manheicatherine**
Hello,
I completed ticket 5, please check! Thanks

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -28,15 +28,59 @@ describe("API Testing", () => {
...
-  describe.only("Ticket 4: GET /api/articles", () => {
+  describe("Ticket 4: GET /api/articles", () => {
     test("returns an array of articles which is sorted by date in descending order ", () => {
       return request(app)
         .get("/api/articles")
         .expect(200)
+        .then(({ body }) => {
+          const objOfArr = body.articles;
+          expect(objOfArr.length).toBe(12);
+        });
```
Comment (alexdevdavis) [inline __tests__/app.test.js:39]: on line 32, your test mentions sorting, so you'll need to use jest-sorted to check it's sorted as you expect

File: __tests__/app.test.js [hunk 2]
```diff
@@ -28,15 +28,59 @@ describe("API Testing", () => {
...
+  });
+
+  describe("Ticket 5: GET /api/articles/:article_id", () => {
+    test("returns an array of article 1", () => {
+      return request(app)
+        .get("/api/articles/1")
+        .expect(200)
+        .then(({ body }) => {
+          const objOfArr = body.articles;
+          expect(objOfArr.length).toBe(1);
```
Comment (alexdevdavis) [inline __tests__/app.test.js]: I'd suggest changing the key name to `article`, as you're only expecting 1 article

File: __tests__/app.test.js [hunk 3]
```diff
@@ -28,15 +28,59 @@ describe("API Testing", () => {
...
+
+          objOfArr.forEach((obj) => {
```
Comment (alexdevdavis) [inline __tests__/app.test.js]: as you're only expecting one thing, you can test directly whether it's the object your expecting, by expecting that object's actual values

File: __tests__/app.test.js [hunk 4]
```diff
@@ -28,15 +28,59 @@ describe("API Testing", () => {
...
+                votes: expect.any(Number),
+                article_img_url: expect.any(String),
+              })
+            );
+          });
+        });
+    });
+    test("Error handling 1: passing a valid but not existing id", () => {
+      return request(app)
+        .get("/api/articles/100")
```
Comment (alexdevdavis) [inline __tests__/app.test.js]: nice 404 test

File: __tests__/app.test.js [hunk 5]
```diff
@@ -28,15 +28,59 @@ describe("API Testing", () => {
...
         .then(({body}) => {
-            const objOfArr = body.articles
-            expect(objOfArr.length).toBe(12);
+          expect(body.msg).toBe("404 Not Found");
         });
     });
+
+    test("Error handling 2: passing a invalid id", () => {
+        return request(app)
+          .get("/api/articles/hellokitty")
```
Comment (alexdevdavis) [inline __tests__/app.test.js]: gotta respect this, `hellokitty`

File: models/articles.models.js [hunk 6]
```diff
@@ -15,5 +15,16 @@ exports.fetchArticles = () => {
       });
       return deleteUnuseItem;
     });
- 
+};
+
+exports.fetchArticlesById = (id) => {
+  return db
+    .query(`SELECT * FROM articles WHERE article_id = ${id};`)
```
Comment (alexdevdavis) [inline models/articles.models.js]: use `$` syntax with a values array to prevent SQL injection

**Review by alexdevdavis — APPROVED** [review]
Nicely done, Catherine approved.
-  prioritise the SQL injection refactor, please
- note some adjustments to make to your tests and naming conventions

File: models/articles.models.js (hunk 6 above)
Comment (manheicatherine) [reply models/articles.models.js]: I made the changes Thanks for your comments!

**Pull request merged**